R/middleware.R

```diff
@@ -341,9 +350,11 @@ HandlerManager <- R6Class("HandlerManager",
           headers$'Content-Type' <- response$content_type
 
           response <- filter(req, response)
-          return(list(status=response$status,
-            body=response$content,
-            headers=headers))
+          return(list(
+            status = response$status,
+            body = if (head_request) "" else response$content,
```

Comment:
In the interim, losing `Content-Length` altogether would be ok. But if it's getting added magically, you might need to do the same digging. :)